src/API/Google/AdsCampaign.php

```diff
@@ -388,6 +403,44 @@ protected function convert_campaign( GoogleAdsRow $row ): array {
 		return $data;
 	}
 
+	/**
+	 * Combine converted campaigns data with campaign criterion results data
+	 *
+	 * @param array $campaigns Campaigns data returned from a query request and converted by convert_campaign function.
+	 *
+	 * @return array
+	 */
+	protected function combine_campaigns_and_campaign_criterion_results( array $campaigns ): array {
+		$campaign_criterion_results = ( new AdsCampaignCriterionQuery() )->set_client( $this->client, $this->options->get_ads_id() )
+			->where( 'campaign.id', array_keys( $campaigns ), 'IN' )
+			// negative: Whether to target (false) or exclude (true) the criterion.
+			->where( 'campaign_criterion.negative', 'false', '=' )
+			->where( 'campaign_criterion.status', 'REMOVED', '!=' )
+			->get_results();
```

Comment:
I added some location criterion directly in https://ads.google.com and I see them showing up. However I noticed that it also returns an additional `campaign_criterion` per campaign. We skip over them if the `$geo_target_constant` is empty, but would it be better to exclude them from the result set?

The best way I can see to do that is to add the following where condition to the query:
```
campaign_criterion.location.geo_target_constant IS NOT NULL
```

Query.php would need to be expanded to support the condition `IS NOT NULL`, but do you think that's worth it to add?
The query would be slightly larger, but we would only get the results we are interested in.

Should be possible to add with something like this (and allow it in validate_compare):
```php
			} elseif ( 'IS NOT NULL' === $compare ) {
				$value = '';
```

Reply:
Nice, I think it's worth adding, it makes the query logic mostly in the SQL-like query. I'll refactor it in the "convert geo constants to county code" PR.